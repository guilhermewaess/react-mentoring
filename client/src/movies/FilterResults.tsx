import * as React from 'react';
import { Col, Container, Row } from 'reactstrap';

interface IProps {
  movies: any[];
}

const sortOptions = [
  'id',
  'title',
  'tagline',
  'vote_average',
  'vote_count',
  'release_date',
  'poster_path',
  'overview',
  'budget',
  'revenue',
];

const orderOptions = [
  'asc',
  'desc'
]

const MoviesFound = (props: IProps) => {
  return <Col xs={{ size: 1, offset: 1 }}> {props.movies.length} Movies found</ Col>;
};

const SortBy = (props: any) => (
  <Col xs="auto">
    Sort by:{' '}
    <select 
      id="sort-by-select"
      className="rounded"
      data-filter-field="sortBy"
      onChange={props.onFilterChange}
      value={props.filter.sortBy}>
      {sortOptions.map((option: string, index: number) => (
        <option key={`${option}-${index}`} value={option}>
          {option}
        </option>
      ))}
    </select>
  </Col>
);

const OrderBy = (props: any) => (
  <Col xs="auto">
    Order by:{' '}
    <select
      className="rounded"
      id="order-by-select"
      data-filter-field="sortOrder"
      onChange={props.onFilterChange}
      value={props.filter.orderBy}>
      {orderOptions.map((option: string, index: number) => (
        <option key={`${option}-${index}`} value={option}>
          {option}
        </option>
      ))}
    </select>
  </Col>
)

export default (props: any) =>
  (<Container fluid={true}>
    <Row className="pt-2 pb-2">
      <MoviesFound movies={props.movies} />
      <Col xs="7" />
      <SortBy filter={props.filter} onFilterChange={props.onFilterChange} />
      <OrderBy filter={props.filter} onFilterChange={props.onFilterChange} />
    </Row>
  </Container>)
