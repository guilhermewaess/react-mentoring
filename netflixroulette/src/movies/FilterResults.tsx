import * as React from 'react';

export class FilterResults extends React.Component {
  public render() {
    return (
      <div className='mb-2 d-flex flex-row justify-content-between'>
        <div>X Movies found</div>
        <div>
          Sort by:
          <select name="" id="">
            <option>Release</option>
          </select>
        </div>
      </div>
    )
  }
}