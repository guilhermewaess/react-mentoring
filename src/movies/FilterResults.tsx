import * as React from 'react';
import { Container, Row } from 'reactstrap';

interface IProps {
  movies: any[];
}

const MoviesFound = (props: IProps) => {
  return <div>{props.movies.length} Movies found</div>;
};

const SortBy = () => (
  <div>
    Sort by:
    <select name="" id="">
      <option>Release</option>
    </select>
  </div>
);

export default (props: any) =>
  (<Container>
    <Row className="justify-content-between pt-2 pb-2">
      <MoviesFound movies={props.movies} />
      <SortBy />
    </Row>
  </Container>)
