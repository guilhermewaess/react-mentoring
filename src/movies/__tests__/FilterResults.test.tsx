import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import FilterResults from './../FilterResults';

describe('Movies FilterResults', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<FilterResults movies={[]} />)
  });

  describe('when construct', () => {
    it('should render component', () => {
      expect(shallowToJson(component)).toMatchSnapshot();
    });
    it('should render MoviesFound properly', () => {
      const moviesFound = component.find('MoviesFound').shallow();
      expect(shallowToJson(moviesFound)).toMatchSnapshot();
    });
    it('should render SortyBy properly', () => {
      const sortBy = component.find('SortBy').shallow();
      expect(shallowToJson(sortBy)).toMatchSnapshot();
    });
  });
});