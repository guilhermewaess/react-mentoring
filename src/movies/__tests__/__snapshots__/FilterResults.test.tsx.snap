// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies FilterResults when construct should render MoviesFound properly 1`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs={
    Object {
      "offset": 1,
      "size": 1,
    }
  }
>
   
  0
   Movies found
</Col>
`;

exports[`Movies FilterResults when construct should render OrderBy properly 1`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs="auto"
>
  Order by:
   
  <select
    className="rounded"
    data-filter-field="sortOrder"
    id="order-by-select"
    onChange={[MockFunction]}
    value="orderBy"
  >
    <option
      key="asc-0"
      value="asc"
    >
      asc
    </option>
    <option
      key="desc-1"
      value="desc"
    >
      desc
    </option>
  </select>
</Col>
`;

exports[`Movies FilterResults when construct should render SortyBy properly 1`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs="auto"
>
  Sort by:
   
  <select
    className="rounded"
    data-filter-field="sortBy"
    id="sort-by-select"
    onChange={[MockFunction]}
    value="desc"
  >
    <option
      key="id-0"
      value="id"
    >
      id
    </option>
    <option
      key="title-1"
      value="title"
    >
      title
    </option>
    <option
      key="tagline-2"
      value="tagline"
    >
      tagline
    </option>
    <option
      key="vote_average-3"
      value="vote_average"
    >
      vote_average
    </option>
    <option
      key="vote_count-4"
      value="vote_count"
    >
      vote_count
    </option>
    <option
      key="release_date-5"
      value="release_date"
    >
      release_date
    </option>
    <option
      key="poster_path-6"
      value="poster_path"
    >
      poster_path
    </option>
    <option
      key="overview-7"
      value="overview"
    >
      overview
    </option>
    <option
      key="budget-8"
      value="budget"
    >
      budget
    </option>
    <option
      key="revenue-9"
      value="revenue"
    >
      revenue
    </option>
  </select>
</Col>
`;

exports[`Movies FilterResults when construct should render component 1`] = `
<Container
  fluid={true}
  tag="div"
>
  <Row
    className="pt-2 pb-2"
    tag="div"
  >
    <MoviesFound
      movies={Array []}
    />
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="7"
    />
    <SortBy
      filter={
        Object {
          "orderBy": "orderBy",
          "sortBy": "desc",
        }
      }
      onFilterChange={[MockFunction]}
    />
    <OrderBy
      filter={
        Object {
          "orderBy": "orderBy",
          "sortBy": "desc",
        }
      }
      onFilterChange={[MockFunction]}
    />
  </Row>
</Container>
`;
